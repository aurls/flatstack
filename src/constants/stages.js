export default {
  SHIPPING: 'Shipping',
  BILLING: 'Billing',
  PAYMENT: 'Payment',
  SUCCESS: 'Success',
  PRINT_RECIPE: 'Print recipe'
};
