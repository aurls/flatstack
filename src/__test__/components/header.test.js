import React from 'react';
import Header from '../../components/header';
import stages from '../../constants/stages';

describe('Header component', () => {
  it('Shipping stage', () => {
    const component = shallow(
      <Header
        currentStage={stages.SHIPPING}
        itemsInCartCount={3} />
    );
    expect(component).toMatchSnapshot();
  });

  it('Success stage', () => {
    const component = shallow(
      <Header
        currentStage={stages.SUCCESS}
        itemsInCartCount={3} />
    );
    expect(component).toMatchSnapshot();
  });
});
