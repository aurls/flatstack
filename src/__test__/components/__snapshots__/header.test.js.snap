// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Shipping stage 1`] = `
<header
  className="header"
>
  <div
    className="header__container"
  >
    <div
      className="header__title"
    >
      <img
        alt=""
        className="header__title-logo"
        src={null}
      />
      <div
        className="header__title-text"
      >
        Frontend Developer Test Task
      </div>
    </div>
    <div
      className="header__cart"
    >
      <div
        className="header__cart-title"
      >
        Cart
      </div>
      <img
        alt=""
        className="header__cart-icon"
        src={null}
      />
      <div
        className="header__cart-count"
      >
        <span>
          3
        </span>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header component Success stage 1`] = `
<header
  className="header"
>
  <div
    className="header__container"
  >
    <div
      className="header__title"
    >
      <img
        alt=""
        className="header__title-logo"
        src={null}
      />
      <div
        className="header__title-text"
      >
        Frontend Developer Test Task
      </div>
    </div>
    <div
      className="header__cart"
    >
      <div
        className="header__cart-title"
      >
        Cart
      </div>
      <img
        alt=""
        className="header__cart-icon"
        src={null}
      />
    </div>
  </div>
</header>
`;
